staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -219,7 +224,7 @@ func (f *FieldManager) Apply(liveObj, appliedObj runtime.Object, manager string,
 	}
 
 	// Decode the managed fields in the live object, since it isn't allowed in the patch.
-	managed, err := internal.DecodeObjectManagedFields(accessor.GetManagedFields())
+	managed, err := DecodeManagedFields(accessor.GetManagedFields())
```

Comment:
Yes, but somewhat unrelated, shouldn't we do something useful if the managedfields is invalid here?

Should we replace `emptyManagedFieldsOnErr` method with a `decodeManagedFieldsOrEmpty`, which could be used here?

Reply:
Shouldn't we keep returning the error here if decoding fails (it is apply)?
What would we want to happen if live is corrupted?
It could only happen if somebody updates it and then applies would fail in the future.  I'd say we then should prevent invalid managedFields from getting persisted, what this PR tries to at least make harder.

Reply:
Right, if the managed fields are corrupted in the apiserver, then someone can't apply anymore. The right thing to do is to restart fresh I would say.